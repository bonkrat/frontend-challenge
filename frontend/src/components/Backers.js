import React from "react";
import ReactTooltip from "react-tooltip";
import * as styles from "./Backers.styles";
import PropTypes from "prop-types";

/**
 * Renders a list of backers.
 */
const Backers = ({ backers, size }) => (
  <div className={styles.backers}>
    {backers.map(({ id, name, avatar }) => (
      <React.Fragment key={id}>
        <ReactTooltip effect="solid" />
        <div
          data-tip={name}
          className={styles.backer(size)}
          style={{
            backgroundImage: `url("http://localhost:4000/avatars/${avatar}")`,
            backgroundPosition: "center",
            backgroundSize: "cover"
          }}
        ></div>
      </React.Fragment>
    ))}
  </div>
);

Backers.defaultProps = {
  backers: [],
  size: 6
};

Backers.propTypes = {
  /**
   * An array of backers to render.
   */
  backers: PropTypes.array,
  /**
   * The size of the avatars (1, 2, 4, 6, 8, 16...)
   */
  size: PropTypes.number
};

export default Backers;
