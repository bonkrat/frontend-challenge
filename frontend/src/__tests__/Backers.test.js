import React from "react";
import { shallow } from "enzyme";
import Backers from "../components/Backers";
import toJson from "enzyme-to-json";

const props = {
  backers: [
    {
      id: "foo",
      name: "fooname",
      avatar: "fooavatar"
    },
    {
      id: "bar",
      name: "barname",
      avatar: "baravatar"
    }
  ]
};

it("renders correctly", () => {
  const wrapper = shallow(<Backers {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
