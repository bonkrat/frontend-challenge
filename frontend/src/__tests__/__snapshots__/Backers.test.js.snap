// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex text-base text-gray-500"
>
  <ReactTooltip
    clickable={false}
    effect="solid"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
  <div
    className="rounded-full h-6 w-6 flex items-center justify-center text-white"
    data-tip="fooname"
    style={
      Object {
        "backgroundImage": "url(\\"http://localhost:4000/avatars/fooavatar\\")",
        "backgroundPosition": "center",
        "backgroundSize": "cover",
      }
    }
  />
  <ReactTooltip
    clickable={false}
    effect="solid"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
  <div
    className="rounded-full h-6 w-6 flex items-center justify-center text-white"
    data-tip="barname"
    style={
      Object {
        "backgroundImage": "url(\\"http://localhost:4000/avatars/baravatar\\")",
        "backgroundPosition": "center",
        "backgroundSize": "cover",
      }
    }
  />
</div>
`;
